WooCommerce/UITestsFoundation/Screens/ExternalAppScreen.swift

```diff
@@ -0,0 +1,28 @@
+import XCTest
+
+public final class ExternalAppScreen {
+
+    public private(set) var app: XCUIApplication!
+
+    public init() {
+        app = XCUIApplication()
+    }
+
+    // To open universal links listed in mocked HTML file
+    public func openUniversalLinkFromSafariApp(link: String) {
+
+        let safari = XCUIApplication(bundleIdentifier: "com.apple.mobilesafari")
+        safari.launch()
+
+        // To accommodate elements on both iPhone and iPad
+        let searchBarElement = UIDevice.current.userInterfaceIdiom == .phone ? "TabBarItemTitle" : "UnifiedTabBarItemView?isSelected=true"
+
+        // Go to Wiremock's HTML file with universal links
+        safari.textFields[searchBarElement].tap()
```

Comment:
> Maybe we can ask @tiagomar for a local test run on iPad Air 5 Gen Sim to see if it happes for him too 🙇

Hey @pachlava and @jostnes , I did face the issue and was able to figure out what was causing it. It happens to be one more 🍎 magic.
It only happens when the test opens Safari and it has a loaded page already, like in a second test run if the previous run had the page loaded. The issue is that the Address tab changes its type from TextField to Button if a page is loaded, breaking the `safari.textFields[searchBarElement].tap()`. ¯\\\_(ツ)\_/¯ The change is not reflected on Accessibility Inspector though, maybe because it would make it too easy to catch. =P

I've found a solution for it with minimal changes and avoiding adding more complexity dealing with both devices. There is a higher level `otherElements` element, `UnifiedTabBar,` that can accept the `tap()` and does not mutate depending on what's loaded. I could also find a higher level `otherElements` element for `iPhone`, `CapsuleNavigationBar?isSelected=true`, so we can rely on the same `tap()` command for both platforms.


<details>
  <summary>Suggested locators strategy</summary>
  <img width="914" alt="image" src="https://user-images.githubusercontent.com/42008628/230969810-1cb20c1e-e296-4bcc-8dbe-3eca840f9151.png">
</details>


```suggestion
        let searchBarElement = UIDevice.current.userInterfaceIdiom == .phone ? "CapsuleNavigationBar?isSelected=true" : "UnifiedTabBar"

        // Go to Wiremock's HTML file with universal links
        safari.otherElements[searchBarElement].tap()
```

Reply:
ugh i missed that button / textfield difference 🥲 thanks for suggesting the new elements! they seem to work in different safari states on both iPhone and iPad, updated in https://github.com/woocommerce/woocommerce-ios/pull/9399/commits/6bdd25eaef2b241d112a39b9471b9beb49638ccf